index.d.ts

```diff
@@ -0,0 +1,31 @@
+/**
+Convert a value to an array.
+
+_Supplying `null` or `undefined` results in an empty array._
+
+@example
+```
+import arrify = require('arrify');
+
+arrify('🦄');
+//=> ['🦄']
+
+arrify(['🦄']);
+//=> ['🦄']
+
+arrify(new Set(['🦄']));
+//=> ['🦄']
+
+arrify(null);
+//=> []
+
+arrify(undefined);
+//=> []
+```
+*/
+declare function arrify(value: null | undefined): [];
+declare function arrify(value: string): [string];
```

Comment:
I don't think you should special-case `string`. It's a separate check in the code as string is also iterable, so otherwise it would be used as an iterable instead of a string.

How about this instead?

```suggestion
declare function arrify<ValueType(value: ValueType): [ValueType];
```

Can you also add a (type and normal) test for `arrify(1)` and `arrify(true)`?

Reply:
You're right about the catch-all case, missed this one. `string` is however iterable, so we still have to special-case it.